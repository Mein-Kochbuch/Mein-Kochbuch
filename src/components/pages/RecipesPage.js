import React from "react";
import Header from "./Header";
import styled from "styled-components/native";
import RecipeList from "../../components/recipes/RecipeList";

export default function RecipesPage({recipes, loadNext, title}) {

    const loadMoreRecipes = () => {
        loadNext()
    }

    return (
        <ViewStyled>
            <Header title={title}/>
            <RecipeList recipes={recipes} loadMoreRecipes={loadMoreRecipes}/>
        </ViewStyled>
    )

}

const ViewStyled = styled.View`
  height: 100%;
`
