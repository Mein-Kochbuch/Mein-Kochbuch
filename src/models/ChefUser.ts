export type ChefUser = {
  id: string;
  name: string;
};
