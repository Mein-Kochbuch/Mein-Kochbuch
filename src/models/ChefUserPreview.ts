export type ChefUserPreview = {
  id: string;
  name: string;
};
