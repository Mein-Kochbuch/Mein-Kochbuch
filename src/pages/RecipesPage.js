import React from "react";
import Header from "../components/Header";
import styled from "styled-components/native";
import useRecipes from "../hooks/useRecipes";
import RecipeList from "../components/RecipeList";

export default function RecipesPage({title}) {
    const {recipes} = useRecipes()

    console.log(recipes[0])

    return (
        <ScrollViewStyled>
            <Header title={title}/>
            <RecipeList recipes={recipes}/>
        </ScrollViewStyled>
    );
}

const ScrollViewStyled = styled.ScrollView`
  height: 100%;
`
