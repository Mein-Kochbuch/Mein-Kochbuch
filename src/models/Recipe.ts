import {Image} from './Image';
import {ChefUser} from './ChefUser';

export interface Recipe {
  id: string;
  title: string;
  owner: ChefUser;
  privacy: boolean;
  instruction: string;
  duration: number;
  difficulty: Difficulty;
  portions: number;
  thumbnail: Image;
  images: Image[];
  ingredients: Ingredient[];
  averageRating: number;
  ratingCount: number;
}

export type Ingredient = {
  id: string;
  amount: number;
  text: string;
};

export enum Difficulty {
  VERY_EASY = 0,
  EASY = 1,
  MEDIUM = 2,
  HARD = 3,
  EXPERT = 4,
}
