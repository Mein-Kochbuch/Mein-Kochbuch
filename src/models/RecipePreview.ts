import {Image} from './Image';
import {ChefUserPreview} from './ChefUserPreview';

export interface RecipePreview {
  id: string;
  title: string;
  averageRating: number;
  thumbnail?: Image;
  ratingCount: number;
  owner: ChefUserPreview;
}
