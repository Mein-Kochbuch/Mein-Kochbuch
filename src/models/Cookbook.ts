import {Image} from './Image';
import {Recipe} from './Recipe';
import {ChefUserPreview} from './ChefUserPreview';

export interface Cookbook {
  id: string;
  name: string;
  owner: ChefUserPreview;
  privacy: boolean;
  recipes: Recipe[];
  thumbnail?: Image;
}
