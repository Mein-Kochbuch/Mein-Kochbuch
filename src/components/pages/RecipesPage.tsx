import React from 'react';
import Header from './Header';
import styled from 'styled-components/native';
import RecipeList from '../recipes/RecipeList';
import {RecipePreview} from '../../models/RecipePreview';

interface RecipePageProps {
  recipes: RecipePreview[];
  loadNext: () => void;
  title: string;
}

export default function RecipesPage({
  recipes,
  loadNext,
  title,
}: RecipePageProps) {
  const loadMoreRecipes = () => {
    loadNext();
  };

  return (
    <ViewStyled>
      <Header title={title} />
      <RecipeList recipes={recipes} loadMoreRecipes={loadMoreRecipes} />
    </ViewStyled>
  );
}

const ViewStyled = styled.View`
  height: 100%;
`;
