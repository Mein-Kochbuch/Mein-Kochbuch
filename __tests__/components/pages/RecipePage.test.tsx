import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import RecipesPage from '../../../src/components/pages/RecipesPage';
import {RecipePreview} from '../../../src/models/RecipePreview';

describe('RecipesPage', () => {
  const recipes: RecipePreview[] = [
    {
      id: '1',
      title: 'test-title-1',
      ratingCount: 3,
      averageRating: 3.8,
      owner: {
        id: '1',
        name: 'me',
      },
      thumbnail: {
        id: '1',
        url: 'thumbnail-url-1',
      },
    },
    {
      id: '2',
      title: 'test-title-2',
      ratingCount: 16,
      averageRating: 4.1,
      owner: {
        id: '2',
        name: 'you',
      },
      thumbnail: {
        id: '2',
        url: 'thumbnail-url-2',
      },
    },
  ];

  it('Snapshot Test', () => {
    const loadNext = () => {};

    const component = renderer.create(
      <RecipesPage
        recipes={recipes}
        loadNext={loadNext}
        title={'test-title'}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
