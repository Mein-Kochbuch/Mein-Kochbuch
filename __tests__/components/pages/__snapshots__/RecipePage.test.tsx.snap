// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesPage Snapshot Test 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
      },
    ]
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "marginVertical": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 50,
          "textAlign": "center",
        }
      }
    >
      test-title
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "height": "88%",
        },
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "averageRating": 3.8,
            "id": "1",
            "owner": Object {
              "id": "1",
              "name": "me",
            },
            "ratingCount": 3,
            "thumbnail": Object {
              "id": "1",
              "url": "thumbnail-url-1",
            },
            "title": "test-title-1",
          },
          Object {
            "averageRating": 4.1,
            "id": "2",
            "owner": Object {
              "id": "2",
              "name": "you",
            },
            "ratingCount": 16,
            "thumbnail": Object {
              "id": "2",
              "url": "thumbnail-url-2",
            },
            "title": "test-title-2",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#222",
                "borderRadius": 10,
                "borderStyle": "solid",
                "borderWidth": 2,
                "display": "flex",
                "flexDirection": "row",
                "height": 120,
                "marginBottom": 12,
                "marginLeft": 12,
                "marginRight": 12,
                "marginTop": 12,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "thumbnail-url-1",
                }
              }
              style={
                Array [
                  Object {
                    "backgroundColor": "#68a0cf",
                    "borderColor": "#fff",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": 100,
                    "paddingBottom": 20,
                    "paddingTop": 20,
                    "width": 100,
                  },
                ]
              }
            />
            <Text
              ellipsizeMode="middle"
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "fontSize": 24,
                    "marginLeft": 12,
                    "maxWidth": "70%",
                    "textAlign": "center",
                  },
                ]
              }
            >
              test-title-1
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#222",
                "borderRadius": 10,
                "borderStyle": "solid",
                "borderWidth": 2,
                "display": "flex",
                "flexDirection": "row",
                "height": 120,
                "marginBottom": 12,
                "marginLeft": 12,
                "marginRight": 12,
                "marginTop": 12,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "thumbnail-url-2",
                }
              }
              style={
                Array [
                  Object {
                    "backgroundColor": "#68a0cf",
                    "borderColor": "#fff",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": 100,
                    "paddingBottom": 20,
                    "paddingTop": 20,
                    "width": 100,
                  },
                ]
              }
            />
            <Text
              ellipsizeMode="middle"
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "fontSize": 24,
                    "marginLeft": 12,
                    "maxWidth": "70%",
                    "textAlign": "center",
                  },
                ]
              }
            >
              test-title-2
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
